File: qubit/kotlin/QubitBuilder.kt
import kotlin.math.pow

/**
 * This class builds several types of qubits
 * @author Paul Rodriguez-Ch
 */
class QubitBuilder {

    companion object {

        /**
         * This method creates a qubit with random probability
         * @return a qubit with random probability
         */
        fun buildRandomQubit(): Qubit {
            var normalizedVector: ArrayList<Float> = Utilities.randomDistributionGeneration(4)
            var alphaFunction: Complex = Complex(normalizedVector[0].pow(0.5f), normalizedVector[1].pow(0.5f))
            var betaFunction: Complex = Complex(normalizedVector[2].pow(0.5f), normalizedVector[3].pow(0.5f))
            return Qubit(alphaFunction, betaFunction)
        }

        /**
         * This method creates a qubit with equal-opportunity
         * @return an equal-opportunity qubit
         */
        fun buildEqualOpportunityQubit(): Qubit {
            var equalOpportunityNumber: Float = 1.toFloat() / 2.toFloat().pow(0.5f)
            var alphaFunction: Complex = Complex(equalOpportunityNumber, 0f)
            var betaFunction: Complex = Complex(equalOpportunityNumber, 0f)
            return Qubit(alphaFunction, betaFunction)
        }
    }
}

File: qubit/python/qubitBuilder.py
from utils import Utilities
from complex import Complex
from qubit import Qubit


class QubitBuilder:
    """
    This class builds several types of qubits 
    """

    @staticmethod
    def build_random_qubit():
        """
        This method creates a qubit with random probability
        :return: qubit with random probability
        """
        normalize_vector = Utilities.random_distribution_generation(4)
        alpha_function = Complex(pow(normalize_vector[0], 0.5), pow(normalize_vector[1], 0.5))
        beta_function = Complex(pow(normalize_vector[2], 0.5), pow(normalize_vector[3], 0.5))
        return Qubit(alpha_function, beta_function)

    @staticmethod
    def build_equal_opportunity_qubit():
        """
        This method creates a qubit with equal-opportunity
        :return: equal-opportunity qubit 
        """
        equal_opportunity_number = 1/pow(2,0.5)
        alpha_function = Complex(equal_opportunity_number, 0)
        beta_function = Complex(equal_opportunity_number, 0)
        return Qubit(alpha_function, beta_function)